include/FileBrowser.h

```diff
@@ -73,7 +73,9 @@ class FileBrowser : public SideBarWidget
 			const QString & title, const QPixmap & pm,
 			QWidget * parent, bool dirs_as_items = false,
 			const QString& userDir = "",
-			const QString& factoryDir = "");
+			const QString& factoryDir = "",
+			bool isFavoritesBrowser = false
```

Comment:
The only reason why I would prefer a new class is if we want to add more functionality in the future specific to the "Favorite" browser to avoid clogging up the original `FileBrowser`.

Reply:
Another solution might be just to use Qt signals to.